### ISSUE TITLE
Can we help keep this repo going?

### ISSUE THREAD

**Issue by donkidd** [m4tthumphrey/php-gitlab-api#481]
There are several PR's out there that look like they could be merged.  @m4tthumphrey can I or any of us help with PR approvals, or is there anything we(the community)  do to help with this repo?

**Comment by m4tthumphrey** [comment]
Hey @donkidd,

I am not very active in this project due to not actually use it much these days. However, @jubianchi, @radutopala, @fbourigault, @m1guelpf are all collaborators. I have no issue adding more if required...

**Comment by donkidd** [comment]
I just see lots of PR's out there with green checkmarks, so just wondering what would need to happen to get them Merged.  Mine PR has been out there 3 weeks, fixed a defect and added a new test, but have seen no response or acknowledgment about what is to happen next.  If you need someone in the US, I could help with things in this time zone.

**Comment by m4tthumphrey** [comment]
@donkidd Would you still like to become a collaborator on this project?

**Comment by donkidd** [comment]
Sure, we are using this a lot for a project I am working on.

**Comment by glensc** [comment]
@donkidd most of the open PR's are stalled because of original reporters either:
1. don't add tests
2. don't rebase

you can help by submitting your own rebased PR, like this one:
- https://github.com/m4tthumphrey/php-gitlab-api/pull/503

that would require no involvement from maintainers.

**Comment by glensc** [comment]
Also, contributors rather close the PR and forget about it than write put extra effort to satisfy the requirements for merge:
- https://github.com/m4tthumphrey/php-gitlab-api/pull/198

**Comment by glensc** [comment]
I also noticed that most of the tests are really rather useless:
1. they build arbitrary $returnArray
2. mock that whatever the API call does, return the $returnArray

so, it's testing the validity of the $returnArray, you can change it to anything and test will still pass. to remove confusion, I would remove such assertions at all, they test that variable equals to itself.

those tests only validate that API calls are made with proper $path and $params, and some API calls would handle OptionsResolve.

here's good comic, illustrating the situation:
- http://geek-and-poke.com/geekandpoke/2019/10/20/qa-best-practices

**Comment by GrahamCampbell** [comment]
@donkidd I have come on board as a maintainer now, and we're going to transfer this repo to a new home, in a GitHub organisation soon. A lot of the PRs are unfortunately stalled because they are not of the required quality, most of the time, because they lack any tests. I'll prompt authors again, and see which I can quickly finish myself.

If anyone else is interested in helping with this process, or would like to draw my attention to any specific issues or PRs, please let met know. Just @ me on the issue in question, and I'll take a look. I plan to release 9.18.0 and 10.0.0 later next week, if everything goes to plan.

**Comment by glensc** [comment]
@GrahamCampbell please see my rant above, requesting the tests is a pointless blocker. as the existing tests do not test actually the API, they just test input arguments, and as the input arguments are also injected, they become totally useless (you're just testing that you receive values you just injected)

**Comment by GrahamCampbell** [comment]
Yeh, I'm aware that the tests are not great, and even return the wrong type sometimes! I want to always return either an array or a StreamInterface (for non-json replies) in 10.0, like https://packagist.org/packages/bitbucket/client does, and enforce this strictly!

We do now have some tests that hit the actual API now, btw, and I've been able to run gitlab locally, so we can possibly make more of the tests run against that, however it takes like 10-15 minutes to boot, so we'd be quite far away from sub-1-second unit test runs, even waiting for our local test instance to boot.

I've looked at all the PRs, and I've merged what I can, but many of the remaining ones have outstanding issues in the source code, and not just the tests.
